### PR TITLE
refactor(core): static query schematic should handle asynchronous query usages properly

### PR DESCRIPTION
**Pull request by devversion**
* See individual commits

### CODE REVIEW

File: packages/core/schematics/migrations/static-queries/typescript/functions.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+/** Checks whether a given node is a function like declaration. */
+export function isFunctionLikeDeclaration(node: ts.Node): node is ts.FunctionLikeDeclaration {
+  return ts.isFunctionDeclaration(node) || ts.isMethodDeclaration(node) ||
+      ts.isArrowFunction(node) || ts.isFunctionExpression(node) ||
+      ts.isGetAccessorDeclaration(node) || ts.isSetAccessorDeclaration(node);
+}
+
+/**
+ * Unwraps a given expression TypeScript node. Expressions can be wrapped within multiple
+ * parenthesises. e.g. "(((({exp}))))()". The function should return the TypeScript node
```
Comment (kara) [inline packages/core/schematics/migrations/static-queries/typescript/functions.ts]:
```suggestion
 * parentheses. e.g. "(((({exp}))))()". The function should return the TypeScript node
```

**Review by kara — APPROVED** [review]
LGTM, sans typo

**Comment by kara** [comment]
[presubmit](http://test/OCL:238063020:BASE:238063095:1552416321361:dc050ab0)

File: packages/core/schematics/migrations/static-queries/typescript/functions.ts (hunk 1 above)
Comment (devversion) [reply packages/core/schematics/migrations/static-queries/typescript/functions.ts]: Good catch. thanks! not sure what this currently is 😁

**Pull request closed without merging**